Install software as used within elgentos

janmartenjongerius opened the pull request:
### Changes

- Google Chrome is now optional

### Additions

- The AWS CLI command `aws` is installed using Docker
- Added Brave Browser
- Added Firefox
- Added Dnsmasq
- Added Dnsmasq configuration for Docker Compose Development
- Added ssg-js

Commit:
Add workspaces as named directories

Commit:
Make Google Chrome optional software

Commit:
Add aws-cli as an alias

Commit:
Add Brave Browser and Firefox

janmartenjongerius commented on .zshrc:
```diff
@@ -18,7 +18,15 @@ plugins=(
 [ -f "$HOME/.zshrc-development" ] && source "$HOME/.zshrc-development"
+
+# Run AWS CLI through Docker.
+alias aws='docker run --rm -it -v ~/.aws:/root/.aws -v $(pwd):/aws amazon/aws-cli'
```
@JeroenBoersma is there anything you personally added on top of this?

JeroenBoersma replied:
Yes I've... 

Here you go!

JeroenBoersma replied:
https://gist.github.com/JeroenBoersma/87e29fd4aa06ec42216c80a6e3649fa5

Commit:
Move development/ to home

Commit:
Allow additional commands after make

By supplying an additional command to the `TINKER` environment variable,
this command is appended to the list of commands executed within the
docker container.

Additionally, the image name is now quoted, to prevent escaping from the
docker command.

Lastly, the `-t` flag has been removed, because this makes the terminal
behave interactively when it shouldn't.

Commit:
Add dnsmasq and docker-compose-development-dnsmasq

Commit:
Do not remove docker volumes as dependency

The docker volumes can be automatically removed if the development setup
is run for the first time.

Since setting it up as dependency does not result in a new file, the
dependency will always run, which is unwanted behavior.

The dependency is merged back within the make target for the development
environment.

Commit:
Replace aws alias with script from Jeroen Boersma

johmanx10 commented:
@JeroenBoersma the following changes are made

- The `aws` alias has been turned into a local installation of your AWS gist, allowing developers to `git pull` for updates, as you announce them.
- Added Dnsmasq
- Added Dnsmasq configuration for Docker Compose Development, which allows DNS resolution of `*.localhost` (or whatever the configured `$DOMAINSUFFIX` is set at within the development repo) within Docker containers.

Commit:
Add ssg-js

Commit:
Do not try to create the SSH config file

The SSH config file is a problematic file in Docker contexts. If the
user somehow is missing the `~/.ssh/config` file, they can manually
repair this.

Commit:
Restore SSH mountpoint

Commit:
Run make as user with the same id and group as the host user.

johmanx10 commented:
I just found out the nameserver configuration does not survive a reboot. Will have to look into that.

Commit:
Add support for NetworkManager

Commit:
Only restart the network manager service if it is installed

Commit:
Remove test against upgrade and autoremove

Commit:
Fall back to GIT clone using HTTPS

Commit:
Add elgentos curated list

Commit:
Only register named directories if their parent exists.

janmartenjongerius commented on Makefile:
```diff
@@ -351,12 +394,149 @@ teamviewer: | $(TEAMVIEWER)
+	echo 'nameserver 9.9.9.9'   | sudo tee -a /etc/resolv.conf
+	sudo mkdir -p /etc/NetworkManager/conf.d
+	echo "[main]\ndns=none\nrc-manager=unmanaged" | sudo tee /etc/NetworkManager/conf.d/dnsmasq.conf
+	sudo service --status-all | awk '{ print $$4 }' | grep -qv network-manager || sudo service network-manager restart
```
Together with Arjen, I observed this command is skipped when it shouldn't be skipped.

Commit:
Allow interactive shell and TTY in `docker-make`

The `aws` command now uses an interactive shell and outputs to TTY.
Because the `docker-make` did not provide a TTY, it could not bind to
one and hence the verification of the `aws --version` command failed
within CI.

By making the shell interactive and using a TTY, the installation for
`docker-compose-development` started asking for user input. This is now
patched, as it provides defaults when INTERACTIVE mode is disabled.

Commit:
Correctly skip network manager restart

Commit:
Do not let ufw depend on file status of bash

Commit:
Allow selectively disabling TTY

Commit:
Fix invalid reference format

JeroenBoersma commented on .zshrc:
```diff
@@ -18,6 +18,11 @@ plugins=(
 for repo in ~/git/*; do hash -d "$(basename "$repo")=$repo"; done
 
+# Make all workspaces into named directories.
+[ -d ~/development/workspace ] && for workspace in ~/development/workspace/*; do
```
you can put your workspace anywhere with `dev volume workspace`

there is a command to retrieve the `workspace` location `dev workspace` which returns it as a string.

JeroenBoersma commented on Makefile:
```diff
@@ -351,12 +397,153 @@ teamviewer: | $(TEAMVIEWER)
+	node \
+	slack \
+	ssg \
+	tmuxinator_completion
```
i would make tmuxinator optional :)

JeroenBoersma commented:
There are a few packages which I think are nice-to-have... (for instance tmuxinator brave(maybe not install them by default, because some of them add complexity too))

JeroenBoersma commented:
I use https://github.com/zsh-users/antigen as a plugin manager for  zsh, I love it, but it's also personal flavor

johmanx10 commented:
@JeroenBoersma your feedback is now covered in https://github.com/elgentos/setup/issues/1
